_P201_VBAN.ino

```diff
@@ -0,0 +1,577 @@
+       GAIN = Unconnected, uncompressed gain set to 60dB. 
+      */
+      if (CONFIG_PIN2>0)
+       switch(PCONFIG(1)){
+          case 1: //40db
+            pinMode(CONFIG_PIN2,OUTPUT);
+            digitalWrite(CONFIG_PIN2,HIGH);
+            break;
+          case 2: //50db
+            pinMode(CONFIG_PIN2,OUTPUT);
+            digitalWrite(CONFIG_PIN2,LOW);
+            break;
+          case 3: //60db
+          default:
+          pinMode(CONFIG_PIN2, INPUT); //high Z
+        }
+
+      /*
+      Trilevel Attack and Release Ratio Select. Controls the ratio of attack time to release time for the AGC circuit.
+       A/R = GND: Attack/Release Ratio is 1:500 
+       A/R = VDD: Attack/Release Ratio is 1:2000 
+       A/R = Unconnected: Attack/Release Ratio is 1:4000
+      */
+      if (CONFIG_PIN3>0)
+        switch(PCONFIG(2)){
+          case 1: //AR=1:500
+            pinMode(CONFIG_PIN3,OUTPUT);
+            digitalWrite(CONFIG_PIN3,LOW);
+            break;
+          case 2: //AR=1:2000
+            pinMode(CONFIG_PIN3,OUTPUT);
+            digitalWrite(CONFIG_PIN3,HIGH);
+            break;
+          case 3: //60db
+          default:
+          pinMode(CONFIG_PIN3, INPUT); //high Z
+        }
+      
+      //p201UDP.begin(VBAN_PORT);
+
+
+      //WiFi.setTxPower(WIFI_POWER_MINUS_1dBm);
+
+      
+      ADC_CHANNEL = (adc1_channel_t)digitalPinToAnalogChannel(CONFIG_PIN1);
+      
+      {
+      adc1_config_width(ADC_WIDTH_12Bit); // configure the analogue to digital converter
+      adc1_config_channel_atten(ADC_CHANNEL, ADC_ATTENUATION); //ADC_ATTEN_11db); // connects the ADC 1 with channel 7 (GPIO 35)
+      adc1_get_raw(ADC_CHANNEL); //we need this to initialize the ADC correctly
+
+      
+      adc1_config_channel_atten(ADC1_CHANNEL_1, ADC_ATTENUATION); //ADC_ATTEN_11db); // connects the ADC 1 with channel 7 (GPIO 35)
+      adc1_get_raw(ADC1_CHANNEL_1); //we need this to initialize the ADC correctly
+      }
+
+      xTaskCreate(adcHandler, "Handler Task", 5000, NULL, 1, &adcHandlerTask);
+
+      //timer = timerBegin(0, 80, true); // 80 Prescaler => 1 microseconds for 8Khz
+      timer = timerBegin(0, 40, true); // 40 Prescaler => 0.5 microseconds  for 16Khz
+      //timer = timerBegin(0, 20, true); // 20 Prescaler => 0.25 microseconds  for 32Khz
+      timerAttachInterrupt(timer, &onTimer, true); // binds the handling function to our timer 
+      timerAlarmWrite(timer, 125, true); //x*8Khz sample rate
+      timerAlarmEnable(timer);
+
+
+      //after the plugin has been initialised successfuly, set success and break
+      log = F("VBAN: Initialized");
+      addLog(LOG_LEVEL_INFO,log);
+      success = true;
+      break;
+
+    }
+
+    case PLUGIN_READ:
+    {
+      //code to be executed to read data
+      //It is executed according to the delay configured on the device configuration page, only once
+
+      //after the plugin has read data successfuly, set success and break
+      success = true;

```

Comment:
This `success = true` statement in `PLUGIN_READ` will cause a set of events (when rules are enabled) every time it is 'executed'.
Not sure if that's what you want on something as time critical as this plugin?

Reply:
You are right is not what I wanted. so success= false. will skip all those or error things out?

Reply:
can I just comment the whole case ?
Both PLUGIN_READ and PLUGIN_WRITE ?

Reply:
> can I just comment the whole case ?
> Both PLUGIN_READ and PLUGIN_WRITE ?

If you don't need them, just comment their cases out.
PLUGIN_WRITE is needed only if you would like to support commands in the plugin.
But I can imagine commands are not needed for this.

Reply:
done.